File: sample.js
function greet(name) {
    console.log(`Hello, ${name}!`);
}

const numbers = [1, 2, 3, 4, 5];
const doubled = numbers.map(x => x * 2);

class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }
    
    introduce() {
        return `Hi, I'm ${this.name} and I'm ${this.age} years old.`;
    }
}

// Arrow function
const add = (a, b) => a + b;

// Async/await example
async function fetchData() {
    try {
        const response = await fetch('/api/data');
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error fetching data:', error);
    }
}

greet('World');

File: sample.rs
fn main() {
    println!("Hello, world!");
    
    let mut x = 42;
    let y = "test string";
    
    if x > 0 {
        println!("x is positive: {}", x);
    }
    
    for i in 0..5 {
        println!("Loop iteration: {}", i);
    }
    
    // This is a comment
    match x {
        42 => println!("The answer!"),
        _ => println!("Something else"),
    }
}

struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: String, age: u32) -> Self {
        Self { name, age }
    }
    
    fn greet(&self) {
        println!("Hello, my name is {}", self.name);
    }
}
